### ISSUE TITLE
npm ERR! code EPERM npm ERR! errno -4048 npm ERR! syscall mkdir npm ERR! Error: EPERM: operation not permitted, mkdir 'C:\Users\user\AppData\Roaming\npm\node_modules\.staging' 

### ISSUE THREAD

**Issue by RamyaKeerthanaPotla** [npm/npm#21058]
This is the error I have encountered while I was trying to install Angular in the terminal of Visual Studio code using the command. 
>npm install -g @angular/cli --save

The version of node using is --> v8.11.3 
Version of npm  ---> 5.6.0

I have been running in the administrator mode of Windows 10, though facing this error.  I couldn't figure out how to fix this bug or find anything related to 'staging' anywhere. 

Thought of clearing the Cacache, but not sure how far would that help with this problem.

Any ideas or help would be appreciated..

Thanks.

npm install -g @angular/cli --save
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: nan@2.10.0 (node_modules\@angular\cli\node_modules\nan):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EPERM: operation not permitted, mkdir 'C:\Users\user\AppData\Roaming\npm\node_modules\.staging'

npm ERR! path C:\Users\user\AppData\Roaming\npm\node_modules\.staging
npm ERR! code EPERM
npm ERR! errno -4048
npm ERR! syscall mkdir
npm ERR! Error: EPERM: operation not permitted, mkdir 'C:\Users\user\AppData\Roaming\npm\node_modules\.staging'
npm ERR!  { Error: EPERM: operation not permitted, mkdir 'C:\Users\user\AppData\Roaming\npm\node_modules\.staging'
npm ERR!   cause:
npm ERR!    { Error: EPERM: operation not permitted, mkdir 'C:\Users\user\AppData\Roaming\npm\node_modules\.staging'
npm ERR!      errno: -4048,
npm ERR!      code: 'EPERM',
npm ERR!      syscall: 'mkdir',
npm ERR!      path: 'C:\\Users\\user\\AppData\\Roaming\\npm\\node_modules\\.staging' },
npm ERR!   stack: 'Error: EPERM: operation not permitted, mkdir \'C:\\Users\\user\\AppData\\Roaming\\npm\\node_modules\\.staging\'',
npm ERR!   errno: -4048,
npm ERR!   code: 'EPERM',
npm ERR!   syscall: 'mkdir',
npm ERR!   path: 'C:\\Users\\user\\AppData\\Roaming\\npm\\node_modules\\.staging',
npm ERR!   parent: '@angular/cli' }
npm ERR!
npm ERR! Please try running this command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Ramya\AppData\Roaming\npm-cache\_logs\2018-06-22T22_58_03_713Z-debug.log

**Comment by RamyaKeerthanaPotla** [comment]
I tried cleaning the cache by the command >npm cache clean This is what it resulted in. 

C:\Users\Ramya\Desktop\UI Training>npm cache clean
npm ERR! As of npm@5, the npm cache self-heals from corruption issues and data extracted from the cache is guaranteed to be valid. If you want to make sure everything is consistent, use 'npm cache verify' instead.
npm ERR!
npm ERR! If you're sure you want to delete the entire cache, rerun this command with --force.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Ramya\AppData\Roaming\npm-cache\_logs\2018-06-23T01_53_19_671Z-debug.log

C:\Users\Ramya\Desktop\UI Training>npm cache verify
Cache verified and compressed (~\AppData\Roaming\npm-cache\_cacache):
Content verified: 512 (14548095 bytes)
Content garbage-collected: 6 (1463988 bytes)
Missing content: 3
Index entries: 783
Finished in 8.539s

**Comment by Idimma** [comment]
Is there a solution yet

**Comment by RamyaKeerthanaPotla** [comment]
No. The command ng is also not working with Visual Studio Code. I don’t
understand whats wrong. It’s a problem with my laptop or OS of my laptop or
with Visual Studio Code or node installations.

This problem is haunting me from May.

**Comment by Idimma** [comment]
You could work around this by deleting the node_modules folder entirely then run **npm install**
or you could remove the read only attribute of the node_modules folder

**Comment by RamyaKeerthanaPotla** [comment]
yeah I did delete the node_modules folder. Then I gave the command 'npm
install' in my cmd(ran as administrator). The result is

**Comment by RamyaKeerthanaPotla** [comment]
Again I tried the command 'npm install ' in Visual studio code. This is the
result




I again tried in my cmd, this is the result

**Comment by Idimma** [comment]
Copy out your project file excluding the node_module into a new folder directory, then run npm install
don't forget your package.json file

**Comment by RamyaKeerthanaPotla** [comment]
I didn't understand '' Copy out your project file excluding the
node_module into a new folder directory, then run npm install
don't forget your package.json file ''

**Comment by RamyaKeerthanaPotla** [comment]
I tried in some other way.  It didn't work. Can you be more specific. My
folder called 'demo project' has no code written in it. That's a newly
created folder. I can entirely delete that one if needed.
